### PR TITLE
Feature/#1792 theme

### PR DESCRIPTION
**Pull request by 1-alex98**
Also wrote a wiki article

### CODE REVIEW

**Comment by 1-alex98** [comment]
https://github.com/FAForever/downlords-faf-client/wiki/Themeing

File: src/main/java/com/faforever/client/preferences/ui/SettingsController.java
Comment (BlackYps) [inline src/main/java/com/faforever/client/preferences/ui/SettingsController.java:245]: when the theme.properties specifies needsRestart=true I get an exception here. I think these are the relevant ones:
`at org.springframework.context.support.AbstractMessageSource.getMessage(AbstractMessageSource.java:153)
	at com.faforever.client.i18n.I18n.get(I18n.java:94)
	at com.faforever.client.i18n.I18n.get(I18n.java:90)
	at com.faforever.client.preferences.ui.SettingsController.lambda$initialize$8(SettingsController.java:245)
[...]
Caused by: java.lang.NumberFormatException: For input string: ""`

**Review by BlackYps — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/preferences/ui/SettingsController.java
Comment (micheljung) [reply src/main/java/com/faforever/client/preferences/ui/SettingsController.java:245]: I propose that we don't leave it up to the theme guy to decide whether or not a restart is neede; he might not even know and it might change in future client releases.

Rather the client should figure out itself whether it's needed or not. In its simplest version: require restart if there's more than just a CSS file.
A future client might be able to reload FXML as well.

Comment (1-alex98) [reply src/main/java/com/faforever/client/preferences/ui/SettingsController.java:245]: If he changes fxml files of pop ups or the gif in the ladder1v1 search it might not be needed even tho it is more than just a css. So I say a maintainer should better judge himself. Also concider context menus, changes there also don't require a restart. You are right that in general none CSS changes more often require a restart but this is not 100%of cases the case.

Comment (1-alex98) [reply src/main/java/com/faforever/client/preferences/ui/SettingsController.java:245]: No answer means u agree with me

Comment (1-alex98) [reply src/main/java/com/faforever/client/preferences/ui/SettingsController.java:245]: Fixed

**Review by BlackYps — DISMISSED** [review]

File: src/main/java/com/faforever/client/theme/Theme.java
Comment (micheljung) [inline src/main/java/com/faforever/client/theme/Theme.java]: I'm still against having such a property.

**Review by micheljung — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/theme/Theme.java
Comment (1-alex98) [reply src/main/java/com/faforever/client/theme/Theme.java]: So what are u for? No restart pop up, always a pop up? Looking if it contains fxml files?(that's inaccurate) Looking if it contains none css files?

File: src/main/java/com/faforever/client/theme/UiService.java
Comment (BlackYps) [inline src/main/java/com/faforever/client/theme/UiService.java]: This will always return the theme.properties file

Comment (1-alex98) [reply src/main/java/com/faforever/client/theme/UiService.java]: What are u talking about? This is a check on wether a theme conatins a file that is not an css file, because if so we need a restart to fully load the theme.

Comment (1-alex98) [reply src/main/java/com/faforever/client/theme/UiService.java]: oh yeah i get u now. Hmm yes so I guess I got to include this there as well

**Pull request merged**
